index.xml

```diff
@@ -3921,6 +3937,102 @@ setlike&lt;<i>type</i>&gt;;</pre>
+            namespace.
+          </p>
+
+          <p>
+            As with interfaces, the IDL for namespaces can be split into multiple parts by using
+            <dfn data-export='' id='dfn-partial-namespace'>partial namespace</dfn> definitions
+            (matching <span class='sym'>"partial"</span> <a class='sym'
+            href='#prod-Namespace'>Namespace</a>). The <a class='dfnref'
+            href='#dfn-identifier'>identifier</a> of a partial namespace definition <span
+            class='rfc2119'>MUST</span> be the same as the identifier of a namespace definition.
+            All of the members that appear on each of the partial namespace definitions are
+            considered to be members of the namespace itself.
+          </p>
+
+          <pre class='syntax'>namespace <em>SomeNamespace</em> {
+  <i>namespace-members…</i>
+};
+
+partial namespace <em>SomeNamespace</em> {
+  <i>namespace-members…</i>
+};</pre>
+
+          <div class='note'>
+            <p>
+              As with partial interface definitions, partial namespace definitions are intended for
+              use as a specification editorial aide, allowing the definition of an interface to be
+              separated over more than one section of the document, and sometimes multiple
+              documents.
+            </p>
+          </div>
+
+          <p>
+            The order that members appear in has no significance except in the case of <a
+            href='#idl-overloading'>overloading</a>.
+          </p>
+
+          <p>
+            Note that unlike interfaces or dictionaries, namespaces do not create types.
+          </p>
+
+          <p>
+            Only the <a class='xattr' href='#Exposed'>[Exposed]</a> and <a class='xattr'
+            href='#SecureContext'>[SecureContext]</a> extended attributes are applicable to
+            namespaces. When declaring a <a class='dfnref' href='#dfn-partial-namespace'>partial
+            namespace</a>, the partial namespace must have the same extended attributes (and
+            extended attribute argument values, if any) specified as the original namespace
+            definition with the same identifier.
+          </p>
+
+          <?productions grammar Partial PartialDefinition Namespace NamespaceMembers NamespaceMember?>
+
+
+          <div class='example'>
+            <p>
+              The following <a class='dfnref' href='#dfn-idl-fragment'>IDL fragment</a> defines an
+              <a class='dfnref' href='#dfn-namespace'>namespace</a>.
+            </p>
+            <x:codeblock language='idl'>namespace VectorUtils {
+  double dotProduct(Vector x, Vector y);
+  Vector crossProduct(Vector x, Vector y);
+};</x:codeblock>
+
+            <p>
+              An ECMAScript implementation would then expose a global property named
+              <code>VectorUtils</code> which was a simple object (with prototype
+              <a>%ObjectPrototype%</a>) with non-enumerable data properties for each declared
```

Comment:
I was torn on this. Safari indicated a desire to match ES's namespaces (Math, Reflect, JSON) which are not. And, `Object.keys(console)` currently returns `[]` because of the prototype separation; this preserves that. To me, it seems more likely people are depending on `Object.keys(console)` being empty (so that e.g. `Object.keys(console)` returns any methods they themselves added), than that they are depending on `Object.getOwnPropertyDescriptor(console, "log").enumerable` being true.

I guess maybe they are using `for`-`in` though, which does change.

I should probably flip this.